Userland/Libraries/LibGUI/FilePicker.cpp

```diff
@@ -297,8 +304,44 @@ void FilePicker::set_path(const String& path)
     }
 
     auto new_path = LexicalPath(path).string();
-    m_location_textbox->set_icon(FileIconProvider::icon_for_path(new_path).bitmap_for_size(16));
     m_model->set_root_path(new_path);
+
+    m_location_combo->set_icon(FileIconProvider::icon_for_path(new_path).bitmap_for_size(16));
+
+    m_path_history.remove_all_matching([&](String const& v) { return v == m_location_combo->text(); });
+    m_path_history.prepend(m_location_combo->text());
+    save_history();
+}
+
+String FilePicker::history_file_path()
+{
+    return LexicalPath::canonicalized_path(String::formatted("{}/{}", Core::StandardPaths::config_directory(), "FilePickerHistory.txt"));
```

Comment:
I think we discussed this on Discord, no history files in .config please :)

Reply:
yea, i wanted to open another PR to implement it for all apps (FilePicker/Run/Shell)
anyway, let's do that :^)